advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -50,19 +51,17 @@ private const val REQUEST_CHUNK_SIZE = 100
 /**
  * A wrapper for [Nexus IQ Server](https://help.sonatype.com/iqserver) security vulnerability data.
  */
-class NexusIq(
-    name: String,
-    config: AdvisorConfiguration
-) : Advisor(name, config) {
+class NexusIq(name: String, private val nexusIqConfig: NexusIqConfiguration) : VulnerabilityProvider(name) {
     class Factory : AbstractVulnerabilityProviderFactory<NexusIq>("NexusIQ") {
-        override fun create(config: AdvisorConfiguration) = NexusIq(advisorName, config)
+        override fun create(config: AdvisorConfiguration): NexusIq {
+            val nexusIqConfig = config.nexusIq
+                ?: throw IllegalArgumentException(
```

Comment:
This check of the specific configuration being present is implemented by all concrete factories. Maybe the base factory class could add support for this, e.g. something like:

```
protected fun <T> getProviderConfiguration(config: AdvisorConfiguration, select: (AdvisorConfiguration) -> T?): T =
  select(config) ?: throw IllegalArgumentException( ... )

```
The `create()` function for NexusIq can then become:

```
override fun create(config: AdvisorConfiguration): NexusIq =
  NexusIq(providerName, getProviderConfiguration(config) { it.nexusIq })
```

Reply:
Good idea!